scripts/set-root.ts

```diff
@@ -1,17 +1,102 @@
 import { program } from "commander";
+import dotenv from "dotenv";
+import { providers, utils } from "ethers";
+import {
+  MerkleDistributor__factory,
+  ExpandedERC20__factory,
+  getDeployedAddress
+} from "@across-protocol/contracts-v2";
 
-// TODO
-// Get inspired from: https://github.com/UMAprotocol/protocol/blob/master/packages/merkle-distributor/scripts/2_PublishClaimsForWindow.ts
+import { parseInputFile } from "../src/fs-utils";
+import {
+  assertAndGetOwnerWallet,
+  checkRecipientAmountsAndDuplicates,
+  ensureApprovedTokens,
+  checkMerkleDistributor
+} from "../src/validation";
+import { publishedTreeFileSchema } from "../src/schemas";
+
+dotenv.config();
 
 program
   .requiredOption(
     "-i, --input <path>",
-    "input JSON file location containing output of 'yarn create-tree'"
+    "input JSON file location containing output of 'yarn publish-tree'"
+  )
+  .option(
+    "-a, --address <address>",
+    "contract address of MerkleDistributor contract"
   )
+  .option("-w, --wait <blocks>", "number of confirmations to wait", "1")
   .parse(process.argv);
 
 async function main() {
-  console.log("TODO");
+  const options = program.opts();
+
+  console.log(`Setting root/window for ${options.input}...`);
+
+  const rawInputFile = parseInputFile(options.input);
+  const inputFile = {
+    ...rawInputFile,
+    // We transform this to an array because yup doesn't have good support for
+    // schemas with dynamic keys and type inference
+    recipientsWithProofs: Object.values(rawInputFile.recipientsWithProofs)
+  };
+
+  console.log("\n1. Validating input file...");
+  const validInputFile = publishedTreeFileSchema.validateSync(inputFile, {
+    abortEarly: false
+  });
+  const merkleDistributorAddress = utils.isAddress(options.address)
+    ? (options.address as string)
+    : getDeployedAddress("MerkleDistributor", validInputFile.chainId);
+  const waitBlocks = Number(options.wait);
+
+  console.log("\n2. Checking recipient amounts and duplicates...");
+  checkRecipientAmountsAndDuplicates(
+    validInputFile.recipientsWithProofs,
+    validInputFile.rewardsToDeposit
+  );
+
+  console.log("\n3. Checking reward token contract...");
+  const { ownerWallet, infuraApiKey } = assertAndGetOwnerWallet();
```

Comment:
Owner wallet should be the owner of the MerkleDistributor, RewardToken or both of them?

Reply:
Ok. Got it. No need to rename